### PR TITLE
Add Clone method to TransactionGateway.

### PR DESCRIPTION
**Pull request by dsmcfarl**
Also added TransactionCloneRequest and supporting test.

### CODE REVIEW

File: transaction_integration_test.go [hunk 1]
```diff
@@ -54,6 +54,25 @@ func TestTransactionCreateSubmitForSettlementAndVoid(t *testing.T) {
...
+	if err != nil {
+		t.Fatal(err)
+	}
+	if x := tx4.Status; x != TransactionStatusSubmittedForSettlement {
+		t.Fatal(x)
+	}
+	if amount := tx4.Amount; amount.Cmp(ten) != 0 {
+		t.Fatalf("transaction settlement amount (%s) did not equal amount requested (%s)", amount, ten)
+	}
+
```
Comment (leighmcculloch) [inline transaction_integration_test.go]: This test is focusing purely on submitting for settlement and then voiding. Can you move the expectations you've added here into a new test that is focused on creating a transaction and cloning it?

File: transaction.go [hunk 2]
```diff
@@ -97,6 +97,30 @@ type TransactionRequest struct {
...
+	MerchantAccountId  string                    `xml:"merchant-account-id,omitempty"`
+	PlanId             string                    `xml:"plan-id,omitempty"`
+	CreditCard         *CreditCard               `xml:"credit-card,omitempty"`
+	Customer           *Customer                 `xml:"customer,omitempty"`
+	BillingAddress     *Address                  `xml:"billing,omitempty"`
+	ShippingAddress    *Address                  `xml:"shipping,omitempty"`
+	TaxAmount          *Decimal                  `xml:"tax-amount,omitempty"`
+	TaxExempt          bool                      `xml:"tax-exempt,omitempty"`
+	DeviceData         string                    `xml:"device-data,omitempty"`
+	Options            *TransactionOptions       `xml:"options,omitempty"`
```
Comment (leighmcculloch) [inline transaction.go]: Comparing the `TransactionsOptions` struct to the [TransactionOptionsCloneRequest](https://github.com/braintree/braintree_java/blob/acc7c9ada19e09f6181dd501657fc7f4f1b7afcb/src/main/java/com/braintreegateway/TransactionOptionsCloneRequest.java) in the Braintree Java SDK, I see that many of the fields in the `TransactionsOptions` struct are not intended to be in a clone options request. 

Can you create a new struct `TransactionCloneOptions` similar to `TransactionOptions` but with a single field `SubmitForSettlement bool`? That's the only field listed in the Braintree Java SDK.

File: transaction.go [hunk 3]
```diff
@@ -97,6 +97,30 @@ type TransactionRequest struct {
...
+	ServiceFeeAmount   *Decimal                  `xml:"service-fee-amount,attr,omitempty"`
+	RiskData           *RiskDataRequest          `xml:"risk-data,omitempty"`
+	Descriptor         *Descriptor               `xml:"descriptor,omitempty"`
+	CustomFields       customfields.CustomFields `xml:"custom-fields,omitempty"`
```
Comment (leighmcculloch) [inline transaction.go]: Comparing this struct to the [TransactionCloneRequest](https://github.com/braintree/braintree_java/blob/acc7c9ada19e09f6181dd501657fc7f4f1b7afcb/src/main/java/com/braintreegateway/TransactionCloneRequest.java) in the Braintree Java SDK, many of these options shouldn't be included in the clone request. The only fields this struct should have are:

* `Amount *Decimal`
* `Channel string`
* `Options TransactionCloneOptions`

**Comment by dsmcfarl** [comment]
@leighmcculloch No problem, thanks for maintaining these bindings. Let me know if the latest changes I pushed don't address your suggestions.

**Comment by leighmcculloch** [comment]
The changes in this PR are in `v0.14.0` which has been released.

**Pull request merged**